crates/block-producer/src/state_view/tests/verify_tx.rs

```diff
@@ -228,3 +229,80 @@ async fn test_verify_tx_vt5() {
         Err(VerifyTxError::InputNotesAlreadyConsumed(vec![nullifier_in_both_txs]))
     );
 }
+
+/// Tests that `verify_tx()` succeeds when the unauthenticated input note found in the in-flight notes
+#[tokio::test]
+#[miden_node_test_macro::enable_logging]
+async fn test_verify_tx_in_flight_notes() {
+    let account_1: MockPrivateAccount<3> = MockPrivateAccount::from(1);
+    let account_2: MockPrivateAccount<3> = MockPrivateAccount::from(2);
+    let store = Arc::new(
+        MockStoreSuccessBuilder::from_accounts(
+            [account_1, account_2]
+                .into_iter()
+                .map(|account| (account.id, account.states[0])),
+        )
+        .build(),
+    );
+    let state_view = DefaultStateView::new(Arc::clone(&store), false);
+
+    let dangling_notes = vec![mock_note(1)];
+    let output_notes = dangling_notes.iter().cloned().map(OutputNote::Full).collect();
+
+    let tx1 = MockProvenTxBuilder::with_account_index(1).output_notes(output_notes).build();
+
+    let verify_tx1_result = state_view.verify_tx(&tx1).await;
+    assert_eq!(verify_tx1_result, Ok(()));
+
+    let tx2 = MockProvenTxBuilder::with_account_index(2)
+        .unauthenticated_notes(dangling_notes.clone())
+        .build();
+
+    let verify_tx2_result = state_view.verify_tx(&tx2).await;
+    assert_eq!(
+        verify_tx2_result,
+        Ok(()),
+        "Dangling unauthenticated notes must be found in the in-flight notes after previous tx verification"
+    );
+}
+
+/// Tests that `verify_tx()` fails when the unauthenticated input note not found not in the in-flight
+/// notes nor in the store
+#[tokio::test]
+#[miden_node_test_macro::enable_logging]
+async fn test_verify_tx_stored_unauthenticated_notes() {
```

Comment:
Nit: test name seems unrelated, maybe something like
```suggestion
async fn test_verify_tx_rejects_unknown_unauthenticated_note() {
```

Reply:
I don't think suggested name is more correct.

Reply:
Naming is hard 😬